geonode/geonodegp/data_queues/processors.py

```diff
@@ -0,0 +1,987 @@
+import csv
+from ftplib import FTP
+import glob
+import json
+import logging
+import urllib
+from geoserver.catalog import Catalog
+import os
+import datetime
+import struct
+import re
+from time import sleep
+from geopy.exc import GeocoderQuotaExceeded
+import requests
+from bs4 import BeautifulSoup as bs
+from geopy.geocoders import GoogleV3
+from pymongo import MongoClient, GEOSPHERE
+from geonodegp.settings import GPM_ACCOUNT, GS_DATA_DIR, HEALTHMAP_AUTH, MONGODB
+from geonodegp.data_queues.csv_helpers import UnicodeWriter, UnicodeReader
+from geonodegp.data_queues.helpers import get_band_count, gdal_translate, cdo_invert, postgres_query, ogr2ogr_exec
+from geonodegp.settings import DEFAULT_WORKSPACE
+from geonode.geoserver.helpers import ogc_server_settings, OGC_Servers_Handler
+from geonode.geoserver.management.commands.updatelayers import Command as UpdateLayersCommand
+import shutil
+
+logger = logging.getLogger("geonodegp.data_queues.processors")
+
+class GeoDataProcessor(object):
```

Comment:
@kotfic  This is a nice set of classes for doing real workflow tasks that could be useful as an example when working on the gaia analysis library.